src/middleware/routing.ts

```diff
@@ -1,10 +1,10 @@
 import Router, { RouterContext } from '@koa/router';
 import createHttpError from 'http-errors';
-import { Middleware, Next } from 'koa';
+import { Middleware, Next, DefaultContext } from 'koa';
 import compose from 'koa-compose';
 
-const notFound = (): Middleware => (
-  async ({ _matchedRoute }: RouterContext, next: Next): Promise<void> => {
```

Comment:
Hmm, `Next` can also be inferred, but had been adding for clarity. Need to find the right balance...

Reply:
I can remove it if you like? Removed the previous one since the type inferred is more complete:
<img width="882" alt="Screenshot 2019-11-18 at 10 57 49" src="https://user-images.githubusercontent.com/8266711/69047103-54f06e00-09f2-11ea-8067-f6d878804560.png">

Reply:
Let's leave both in for now, with consistency with the others. Can have a think separately about it.